pkg/generic/generic.go

```diff
@@ -185,6 +185,24 @@ func SetBinaryWithByteSlice(g Generic, enable bool) error {
 	return nil
 }
 
+// SetSetFieldsForEmptyStruct enable/disable set all fields in map-generic even if a struct is empty.
+//
+//	mode == 0 means disable
+//	mode == 1 means only set required and default fields
+//	mode == 2 means set all fields
+func SetSetFieldsForEmptyStruct(g Generic, mode uint8) error {
```

Comment:
Maybe it needs to clarify that this is for read (decoding)?

Reply:
good idea